tests/functional_tests/ffront_tests/test_execution.py

```diff
@@ -223,6 +225,41 @@ def fencil(out: Field[[Edge], float64], inp: Field[[Edge], float64], factor: flo
     assert np.allclose(ref, out.array())
 
 
+def test_astype(fieldview_backend):
+    size = 10
+    b_float_64 = np_as_located_field(IDim)(np.ones((size), dtype=np.float64))
+    c_bool = np_as_located_field(IDim)(np.ones((size,), dtype=bool))
+    c_int64 = np_as_located_field(IDim)(np.ones((size,), dtype=np.int64))
+    c_int32 = np_as_located_field(IDim)(np.ones((size,), dtype=np.int32))
+    out_bool = np_as_located_field(IDim)(np.zeros((size,), dtype=bool))
+    out_int_64 = np_as_located_field(IDim)(np.zeros((size,), dtype=np.int64))
+    out_int_32 = np_as_located_field(IDim)(np.zeros((size,), dtype=np.int32))
+
+    @field_operator(backend=fieldview_backend)
+    def astype_fieldop_int(b: Field[[IDim], float64]) -> Field[[IDim], int64]:
+        d = astype(b, int64)
+        return d
+
+    astype_fieldop_int(b_float_64, out=out_int_64, offset_provider={})
+    assert np.allclose(c_int64.array(), out_int_64)
+
+    @field_operator(backend=fieldview_backend)
+    def astype_fieldop_bool(b: Field[[IDim], float64]) -> Field[[IDim], bool]:
+        d = astype(b, bool)
+        return d
+
+    astype_fieldop_bool(b_float_64, out=out_bool, offset_provider={})
+    assert np.allclose(c_bool, out_bool)
+
+    @field_operator(backend=fieldview_backend)
+    def astype_fieldop_float(b: Field[[IDim], int64]) -> Field[[IDim], int32]:
+        d = astype(b, int32)
+        return d
+
+    astype_fieldop_float(c_int64, out=out_int_32, offset_provider={})
+    assert np.allclose(c_int32.array(), out_int_32)
```

Comment:
I think it should be separate test cases, as failure of one will not be reported nicely.